raymon/profiling/extractors/vision/anomaly.py

```diff
@@ -1,70 +1,84 @@
-        sample = self.loaded_data[idx]
-        if self.transform:
-            sample = self.transform(sample)
-        return sample
+model_path = pkg_resources.resource_filename("raymon", "models/mobilenetv2-7.onnx")
 
 
 class DN2AnomalyScorer(KMeansOutlierScorer):
     def __init__(self, k=16, size=None, clusters=None, dist="euclidean"):
         super().__init__(k=k, clusters=clusters, dist=dist)
-        self.mobilenet = models.mobilenet_v2(pretrained=True).eval()
+        # model link - https://github.com/onnx/models/tree/master/vision/classification/mobilenet
+        self.mobilenet = onnxruntime.InferenceSession(model_path)
         self.size = size
-        tfs = [
-            transforms.ToTensor(),
-            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
-        ]
-        if size is not None:
-            tfs.append(
-                transforms.Resize(size=size),
-            )
-        self.tfs = transforms.Compose(tfs)
+                
+    def preprocess(self, img):
+        ''' Convert pil_image into numpy,
+            Resize image 
+            Reshape image for onnx model
+        '''
+        numpy_img = np.array(img, dtype=np.float32) 
+        numpy_img = np.resize(numpy_img, (224, 224, 3))
+        numpy_img = numpy_img.reshape([1, 3, 224, 224]) 
+        return numpy_img
+    
+    def prepare_batch(self, images, batch_size):
+        # Our data is a list containing pil images
+        # Convert this image to numpy
+        # Make batches from data and store them in dataloader
+        dataloader = []
+        for i in range(0, len(images), batch_size):
+            # if we arrive the end of the list, we have to adjust the shape of the array,
+            if i + batch_size > len(images):
+                batch_numpy = np.zeros((len(images) - i, 3, 224, 224))
+            # Otherwise the shape of the array is (batch_size, 3, 224, 224).
+            else:
+                # Create zero numpy array with batch shape
+                batch_numpy = np.zeros((batch_size, 3, 224, 224))
+            # Make batches
+            for k in range(i, i + batch_size):
+                # Avoid to exceed the length of the images (list index out of range)
+                if k == len(images):
+                    break
+                x = 0 
+                # Convert pil image and do other preprocessings
+                numpy_img = self.preprocess(images[k])
+                batch_numpy[x, :, :, :] = numpy_img
+                x += 1 
+            dataloader.append(batch_numpy)
+        return dataloader
 
     def extract(self, data):
-        if not isinstance(data, Image.Image):
-            raise ValueError(f"data must be of type PIL.Image.Image, not {data.shape}")
-        batchtf = self.tfs(data)[None, :]
-        feats = self.mobilenet(batchtf).detach().numpy()
-        return super().extract(data=feats)
+        if not isinstance(data, Image.Image) and not isinstance(data, np.ndarray):
```

Comment:
I think `not (isinstance(x, Image) or isinstance(x, numpy))` is ceasier to understand than what is here now. Please change it.